hummingbot/client/performance.py

```diff
@@ -52,210 +54,236 @@ def __init__(self):
+
+    def _preprocess_trades_and_group_by_type(self, trades: List[Any]) -> Tuple[List[Any], List[Any]]:
+        buys = []
+        sells = []
+        for trade in trades:
+            if trade.trade_type.upper() == TradeType.BUY.name.upper():
+                buys.append(trade)
+                self.b_vol_base += Decimal(str(trade.amount))
+                self.b_vol_quote += Decimal(str(trade.amount * trade.price)) * Decimal("-1")
+            elif trade.trade_type.upper() == TradeType.SELL.name.upper():
+                sells.append(trade)
+                self.s_vol_base += Decimal(str(trade.amount)) * Decimal("-1")
+                self.s_vol_quote += Decimal(str(trade.amount * trade.price))
+
+        self.tot_vol_base = self.b_vol_base + self.s_vol_base
+        self.tot_vol_quote = self.b_vol_quote + self.s_vol_quote
+
+        self.avg_b_price = self.divide(self.b_vol_quote, self.b_vol_base)
+        self.avg_s_price = self.divide(self.s_vol_quote, self.s_vol_base)
+        self.avg_tot_price = self.divide(abs(self.b_vol_quote) + abs(self.s_vol_quote),
+                                         abs(self.b_vol_base) + abs(self.s_vol_base))
+        self.avg_b_price = abs(self.avg_b_price)
+        self.avg_s_price = abs(self.avg_s_price)
+
+        return buys, sells
+
+    async def _calculate_fees(self, exchange: str, quote: str, trades: List[Any]):
+        for trade in trades:
+            if self._is_trade_fill(trade):
+                if trade.trade_fee.get("percent") is not None and trade.trade_fee["percent"] > 0:
+                    if quote not in self.fees:
+                        self.fees[quote] = s_decimal_0
+                    self.fees[quote] += Decimal(trade.price * trade.amount * trade.trade_fee["percent"])
+                for flat_fee in trade.trade_fee.get("flat_fees", []):
+                    if flat_fee["asset"] not in self.fees:
+                        self.fees[flat_fee["asset"]] = s_decimal_0
+                    self.fees[flat_fee["asset"]] += Decimal(flat_fee["amount"])
+            else:  # assume this is Trade object
+                if trade.trade_fee.percent > 0:
+                    if quote not in self.fees:
+                        self.fees[quote] = s_decimal_0
+                    self.fees[quote] += (trade.price * trade.order_amount) * trade.trade_fee.percent
+                for flat_fee in trade.trade_fee.flat_fees:
+                    if flat_fee[0] not in self.fees:
+                        self.fees[flat_fee[0]] = s_decimal_0
+                    self.fees[flat_fee[0]] += flat_fee[1]
+
+        for fee_token, fee_amount in self.fees.items():
+            if fee_token == quote:
+                self.fee_in_quote += fee_amount
+            else:
+                last_price = await get_last_price(exchange, f"{fee_token}-{quote}")
+                if last_price is not None:
+                    self.fee_in_quote += fee_amount * last_price
```

Comment:
That is an excellent point Victor, thank you. I think it would be better if we create a separate task for that change. Since it is a functionality change, adding it in a refactoring commit could end up hiding it.